knimin/static/js/platemapper.js

```diff
@@ -269,36 +303,137 @@ PlateMap.prototype.keypressWell = function (current, e) {
 
 /**
  *
+ * Updates the GUI to show the technical replicates
+ *
+ * @param {Object} sampleInfo The sample information object
+ *
+ **/
+PlateMap.prototype._updateTechnicalReplicate = function (sampleInfo) {
+  if (sampleInfo.wells.length > 1) {
+    // Update the list of technical replicates
+    if ($.inArray(sampleInfo, this.technicalReplicates) == -1) {
+      this.technicalReplicates.push(sampleInfo);
+    }
+    // Mark all wells as technical
+    for (var well of sampleInfo.wells) {
+      this.wellInformation[well[0]][well[1]].inputTag.addClass('pm-technical-replicate');
+    }
+  } else {
+    // Using a for loop to avoid to special case - If the wells list is
+    // empty, this for loop will not execute anything. If there is one element
+    // it will correctly remove the mark for techincal replicates
+    for (var well of sampleInfo.wells) {
+      this.wellInformation[well[0]][well[1]].inputTag.removeClass('pm-technical-replicate');
+    }
+    // Remove from the techincal replicates list
+    this.technicalReplicates = $.grep(this.technicalReplicates, function(value) {
+      return value != sampleInfo;
```

Comment:
The return value is used by [JQuery's grep function](http://api.jquery.com/jquery.grep/).